src/charm.py

```diff
@@ -383,7 +383,7 @@ def _render_manifest(self) -> ConfigFileSystemState:
                 self._templates_path: config_suite.templates,
                 self._amtool_config_path: config_suite.amtool,
                 self._server_cert_path: self.server_cert.server_cert,
-                self._key_path: self.server_cert.private_key,
+                self._key_path: self.server_cert.private_key if self.server_cert.enabled else None,
```

Comment:
Do we still need this change?

Reply:
yes
without it, private key will be generated and copied in the container filesytsem without even linking to certs
Also, UTs will break :D

Reply:
Should we do the same about the other paths then?

Reply:
private_key is always present, it should just not be used as an existence check